solidity/contracts/Router.sol

```diff
@@ -194,31 +202,33 @@ abstract contract Router is HyperlaneConnectionClient, IMessageRecipient {
         uint256 _gasPayment,
         address _gasPaymentRefundAddress
     ) internal returns (bytes32 _messageId) {
-        _messageId = _dispatch(_destinationDomain, _messageBody);
-        // Call the IGP even if the gas payment is zero. This is to support on-chain
-        // fee quoting in IGPs, which should always revert if gas payment is insufficient.
-        interchainGasPaymaster.payForGas{value: _gasPayment}(
-            _messageId,
-            _destinationDomain,
+        // Ensure that destination chain has an enrolled router.
+        bytes32 _router = _mustHaveRemoteRouter(_destinationDomain);
+        bytes memory metadata = GlobalHookMetadata.formatMetadata(
+            0,
             _gasAmount,
-            _gasPaymentRefundAddress
+            _gasPaymentRefundAddress,
+            bytes("")
+        );
+        _messageId = mailbox.dispatch{value: _gasPayment}(
+            _destinationDomain,
+            _router,
+            _messageBody,
+            metadata
         );
     }
 
-    /**
-     * @notice Dispatches a message to an enrolled router via the provided Mailbox.
-     * @dev Does not pay interchain gas.
-     * @dev Reverts if there is no enrolled router for _destinationDomain.
-     * @param _destinationDomain The domain of the chain to which to send the message.
-     * @param _messageBody Raw bytes content of message.
-     */
     function _dispatch(uint32 _destinationDomain, bytes memory _messageBody)
```

Comment:
Not sure if this is what Nam called out, but should there be versions of dispatch for all the different `dispatch` variants found on the Mailbox? e.g. specifying your own metadata or hook

Reply:
@yorhodes